mcts.py

```diff
@@ -238,10 +238,12 @@ def is_done(self):
         return self.position.is_game_over() or self.position.n >= FLAGS.max_game_length
 
     def inject_noise(self):
-        dirch = np.random.dirichlet(
-            [FLAGS.dirichlet_noise_alpha] * ((go.N * go.N) + 1))
+        epsilon = 1e-5
+        legal_moves = (1 - (self.illegal_moves / 1000)) + epsilon
```

Comment:
this is sort of weird to recompute the legal moves from the illegal moves, but I guess I don't care about inefficiency since d-noise injection only happens once per move.

Can you abstract out the 1000 as a constant? I chose 1000 because in the worst case, if an illegal move is given prior of 1.0, then it will take ~1,000,000 searches at a node before the action score brings up an illegal move as the top action score (due to the sqrt(parent-N) term in the U calculation), and I figured 1,000,000 searches was probably not going to happen with the python impl, given memory constraints. 

(this reminds me that the C++ impl might actually be able to hit 1,000,000 searches and so 1000 is no longer a suitable value for this constant) FYI @tommadams